src/Services/CoinSelectionService.cs

```diff
@@ -113,13 +115,58 @@ private IBitcoinRequestRepository GetRepository(BitcoinRequestType requestType)
     public async Task LockUTXOs(List<UTXO> selectedUTXOs, IBitcoinRequest bitcoinRequest, BitcoinRequestType requestType)
     {
         // We "lock" the PSBT to the channel operation request by adding to its UTXOs collection for later checking
-        var utxos = selectedUTXOs.Select(x => _mapper.Map<UTXO, FMUTXO>(x)).ToList();
+        var selectedFMUTXOs = selectedUTXOs.Select(x => _mapper.Map<UTXO, FMUTXO>(x)).ToList();
+        var utxos = await _fmutxoRepository.GetByWalletId(bitcoinRequest.Wallet.Id);
 
-        var addUTXOsOperation = await GetRepository(requestType).AddUTXOs(bitcoinRequest, utxos);
+        var utxosToSave = new List<FMUTXO>();
+        var utxosToAddToTransaction = new List<FMUTXO>();
+        // We check the selected UTXOs against the ones we have in the database to see if they are already there
+        foreach (var fmutxo in selectedFMUTXOs)
+        {
+            if (utxos.Contains(fmutxo))
+            {
+                // If they are in the DB we add them to the transaction
+                utxosToAddToTransaction.Add(fmutxo);
+            }
+            else
+            {
+                // If they are not in the DB we add them
+                fmutxo.WalletId = bitcoinRequest.Wallet.Id;
+                fmutxo.SetCreationDatetime();
+                fmutxo.SetUpdateDatetime();
+                utxosToSave.Add(fmutxo);
+            }
+        }
+
+        // If there's any UTXO to save we do it
+        if (utxosToSave.Any())
+        {
+            (bool success, string? reason) = await _fmutxoRepository.AddRangeAsync(utxosToSave);
+            if (!success)
+            {
+                _logger.LogError($"Could not save the utxos that were not selected for request:{bitcoinRequest.Id}");
+                utxosToAddToTransaction = new List<FMUTXO>();
+            }
+
+            foreach (var fmutxo in utxosToSave)
+            {
+                // We get the final UTXO from the DB to add to the transaction
```

Comment:
nit: this is the reason, right?
```suggestion
                // We refetch the UTXOs from the DB so we can have them with the id
```